services/ui_backend_service/data/cache/client/cache_store.py

```diff
@@ -170,42 +169,43 @@ def ensure_path(self, path):
 
     def open_tempdir(self, token, action_name, stream_key):
         self._gc_objects()
+
         if self.total_size > self.max_size:
-            raise CacheFullException("Cache full! Used %d bytes, max %s bytes"
-                                     % (self.total_size, self.max_size))
+            self.warn(None, "Cache soft limit reached! Used %d bytes, max %s bytes"
+                      % (self.total_size, self.max_size))
+
+        try:
+            tmp = tempfile.mkdtemp(prefix='cache_action_%s.' % token,
+                                   dir=self.tmproot)
+        except Exception as ex:
+            msg = "Could not create a temp directory for request %s" % token
+            self.warn(ex, msg)
         else:
-            try:
-                tmp = tempfile.mkdtemp(prefix='cache_action_%s.' % token,
-                                       dir=self.tmproot)
-            except Exception as ex:
-                msg = "Could not create a temp directory for request %s" % token
-                self.warn(ex, msg)
-            else:
-                if stream_key:
-                    src = stream_path(self.root, stream_key)
-                    dst = os.path.join(tmp, key_filename(stream_key))
-                    # make sure that the new symlink points at a valid (empty!)
-                    # file by creating a dummy destination file
-                    self.ensure_path(src)
-                    open_res = self.safe_fileop(open, dst, 'w')
-                    if open_res:
-                        _, f = open_res
-                        f.close()
-                        try:
-                            os.symlink(dst, src)
-                        except OSError as ex:
-                            # two actions may be streaming the same object
-                            # simultaneously. We don't consider an existing
-                            # symlink (errno 17) to be an error.
-                            if ex.errno != 17:
-                                err = "Could not create a symlink %s->%s"\
-                                      % (src, dst)
-                                self.warn(ex, err)
-                        else:
-                            self.active_streams[tmp] = src
-                        return tmp
-                else:
+            if stream_key:
+                src = stream_path(self.root, stream_key)
+                dst = os.path.join(tmp, key_filename(stream_key))
+                # make sure that the new symlink points at a valid (empty!)
+                # file by creating a dummy destination file
+                self.ensure_path(src)
+                open_res = self.safe_fileop(open, dst, 'w')
+                if open_res:
+                    _, f = open_res
+                    f.close()
+                    try:
+                        os.symlink(dst, src)
+                    except Exception as ex:
```

Comment:
originally caught only OSError, as Exception does not have `errno`. Not sure if symlink can raise anything except OSError though, so this might not be an issue

Reply:
Good catch, this is most likely due to older implementation which originated from the cache fork.